include/TFEL/Material/Eshelby.ixx

```diff
@@ -0,0 +1,283 @@
+		{
+			// if (abs(a-b)<value(eps)){
+			//	the formulae must be adapted in the limit case a->b
+			//			   }
+			// if (abs(b-c)<value(eps)){
+			//	the formulae must be adapted in the limit case b->c
+			//			   }
+			// if (abs(a-c)<value(eps)){
+			//	the formulae must be adapted in the limit case a->b->c
+			//			   }
+			// Il faudrait vérifier les formules d'Eshelby ci-dessous
+			const auto a2=a*a;
+		    	const auto b2=b*b;
+		    	const auto c2=c*c;
+		    	const auto Q = 3./8/M_PI/(1-nu);
+		    	const auto R = (1.-2*nu)/8/M_PI/(1-nu);
+		    	const auto k = sqrt(a2-b2)/sqrt(a2-c2);
+		    	const auto theta = asin(sqrt(1-c2/a2));
+		    	const auto F = std::ellint_1(k,theta);
+		    	const auto E = std::ellint_2(k,theta);
+		    	
+		    	const auto Ia = 4.*M_PI*a*b*c/(a2-b2)/sqrt(a2-c2)*(F-E);
+		    	const auto Ic = 4.*M_PI*a*b*c/(b2-c2)/sqrt(a2-c2)*(b*sqrt(a2-c2)/a/c-E);
+		    	const auto Ib = 4.*M_PI-Ia-Ic;
+			const auto Iab = (Ib-Ia)/3/(a2-b2);
+			const auto Iac = (4.*M_PI/3-Ia-(a2-b2)*Iab)/(a2-c2);
+			const auto Ibc = (4.*M_PI/3-Ic-(c2-a2)*Iac)/(c2-b2);
+			const auto Iaa = 4.*M_PI/3/a2-Iab-Iac;
+			const auto Ibb = 4.*M_PI/3/b2-Iab-Ibc;
+			const auto Icc = 4.*M_PI/3/c2-Ibc-Iac;
+			
+			const auto S11 = Q*a2*Iaa+R*Ia;
+			const auto S12 = Q*b2*Iab-R*Ia;
+			const auto S44 = 2.*Q/2*(a2+b2)*Iab+R/2*(Ia+Ib);
+			const auto S22 = Q*b2*Ibb+R*Ib;
+			const auto S21 = Q*a2*Iab-R*Ib;
+			const auto S13 = Q*c2*Iac-R*Ia;
+			const auto S31 = Q*a2*Iac-R*Ic;
+			const auto S33 = Q*c2*Icc+R*Ic;
+			const auto S23 = Q*c2*Ibc-R*Ib;
+			const auto S32 = Q*b2*Ibc-R*Ic;
+			const auto S55 = 2.*Q/2*(a2+c2)*Iac+R/2*(Ia+Ic);
+			const auto S66 = 2.*Q/2*(b2+c2)*Ibc+R/2*(Ib+Ic);
+			
+			const auto zero = real{0};
+			return {S11, S12, S13, 	  zero, zero, zero,
+				S21, S22, S23, 	  zero, zero, zero,
+				S31, S32, S33, 	  zero, zero, zero,
+				zero, zero, zero, S44,  zero, zero,
+				zero, zero, zero, zero, S55,  zero,
+				zero, zero, zero, zero, zero, S66};
+		};//end EshelbyTensorGeneral
+         
+  	}; //end of struct ComputeEshelbyTensor ;
+  	
+  	
+  /*
+  * A simple function that takes a,b,c and returns the indices of the values (a,b,c) sorted from the biggest to the smallest 
+  */
+  template <typename real>
+  const std::array<int,3> sort_ind(const real& a, const real& b, const real& c)
+  {
+	if ((a>b) and (a>c))
+	{
+		if (b>c) return {0,1,2};
+		else return {0,2,1};
+	}
+	else if ((b>a) and (b>c))
+	{
+		if (a>c) return {1,0,2};
+		else return {1,2,0};
+	}
+	else
+	{
+		if (a>b) return {2,0,1};
+		else return {2,1,0}; 
+    	};
+  };//end of sort_ind
+
+ /*
+  * A simple function that takes two vectors and returns the vectorial product
+  */
+  template <typename real>
+  const std::vector<real> VectProd(const std::vector<real>& u, const std::vector<real>& v)
+  {return {u[1]*v[2]-u[2]*v[1],u[2]*v[0]-u[0]*v[2],u[0]*v[1]-u[1]*v[0]};};//end of VectProd
```

Comment:
there is the `cross_product` function in `TFEL/Math`

Reply:
done